cmd/kyma/alpha/deploy/cmd.go

```diff
@@ -124,17 +129,47 @@ func (cmd *command) deploy(start time.Time) error {
 	}
 
 	deployStep := cmd.NewStep("Deploying Kyma")
+	deployStep.Start()
 
-	if err := deploy.Operators(cmd.opts.Source, cmd.K8s, false); err != nil {
+	hasKyma, err := deploy.Bootstrap(cmd.opts.Kustomizations, cmd.K8s, false)
+	if err != nil {
+		deployStep.Failuref("Failed to deploy Kyma.")
 		return err
 	}
-	var err error
 
-	if err != nil {
-		deployStep.Failuref("Failed to deploy Kyma.")
+	// wait for operators to be ready
+	if err := cmd.waitForOperators(); err != nil {
 		return err
 	}
 
+	// deploy kyma CR and modules
+	if hasKyma {
+		kymaStep := cmd.NewStep("Kyma CR deployed")
+		if err := deploy.Kyma(cmd.K8s, false); err != nil {
+			kymaStep.Failuref("Failed to deploy Kyma CR")
+			return err
+		}
+		kymaStep.Success()
+
+		// TODO change to fetch templates from release artifacts
+		modStep := cmd.NewStep("Modules deployed")
+		for _, t := range cmd.opts.Templates {
+			b, err := os.ReadFile(t)
+			if err != nil {
+				modStep.Failuref("Failed to deploy modules")
+				return err
+			}
+			if err := cmd.K8s.Apply(b); err != nil {
+				modStep.Failuref("Failed to deploy modules")
+				return err
+			}
+		}
+		modStep.Success()
```

Comment:
Wouldn't it make more sense to first deploy the templates and then the KymaCR? 

I see that the KymaCR that you are deploying is hard-coded and has no modules, but what if it would be configurable (suggested in another comment by me), then some Errors in the deployed managers would pop it, since some templates cannot be found.

Reply:
I think the deploy order for sure makes sense, however the Kyma we install by default will be empty. The planned way to enable modules in the future is either via busola or via dedicated add/remove module cmd

Reply:
I assumed that was the correct order to ensure a correct deployment. Will turn it around in case a user wants to activate certain modules on deploy.

And of course, as @jakobmoellersap mentions, the default one will always be empty and then modules can be enabled/disabled with other commands. But it is good to have the option to enable N modules while deploying in one go (specially for CI/CD)